File: articles/decision-under-uncertainty-and-computational-risk/c/decision_risk_example.c
#include <stdio.h>

double expected_net_value(double p, double benefit, double loss, double cost) {
    if (p < 0.0) p = 0.0;
    if (p > 1.0) p = 1.0;
    return p * benefit - p * loss - cost;
}

int main(void) {
    printf("%.6f
", expected_net_value(0.42, 150.0, 80.0, 25.0));
    return 0;
}

File: articles/decision-under-uncertainty-and-computational-risk/haskell/DecisionRiskExample.hs
module Main where

expectedNetValue :: Double -> Double -> Double -> Double -> Double
expectedNetValue probability benefit loss cost = bounded * benefit - bounded * loss - cost
  where bounded = max 0.0 (min 1.0 probability)

main :: IO ()
main = print (expectedNetValue 0.42 150.0 80.0 25.0)
